ee/secureenclavesigner/secureenclavesigner.go

```diff
@@ -0,0 +1,230 @@
+//go:build darwin
+// +build darwin
+
+package secureenclavesigner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"os/user"
+	"strings"
+	"time"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/kolide/launcher/ee/allowedcmd"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+type SignRequest struct {
+	Request
+	// Digest is the hash []byte of the data to be signed
+	Digest []byte `msgpack:"digest"`
+	// SecureEnclavePubKey is the B64 encoded DER of the public key to be used to verify the signature
+	SecureEnclavePubKey []byte `msgpack:"secure_enclave_pub_key"`
+}
+
+type Request struct {
+	// Challenge is the B64 encoded challenged generated by the server
+	Challenge []byte `msgpack:"challenge"`
+	// ServerPubKey is the B64 encoded DER of the public key
+	// to be used to verify the signature of the request
+	ServerPubKey []byte `msgpack:"server_pub_key"`
+}
+
+type opt func(*secureEnclaveSigner)
+
+// WithExistingKey allows you to pass the public portion of an existing
+// secure enclave key to use for signing
+func WithExistingKey(publicKey *ecdsa.PublicKey) opt {
+	return func(ses *secureEnclaveSigner) {
+		ses.pubKey = publicKey
+	}
+}
+
+type secureEnclaveSigner struct {
+	uid                  string
```

Comment:
Alternatively, wondering if we might want to do `u, err := user.LookupId(ses.uid)` inside `New` instead to perform UID validation early on, and then we can set the user instead of the UID in this struct instead.

Reply:
not sure on naming either, but I did add the look up to new and added comments on struct fields

Reply:
I like how you updated it, this works for me 👍